src/main/java/org/folio/helper/CheckinHelper.java

```diff
@@ -94,16 +94,10 @@ private Future<Map<String, List<Piece>>> createItemsWithPieceUpdate(CheckinColle
             var srcTenantId = piece.getReceivingTenantId();
             var dstTenantId = checkInPiece.getReceivingTenantId();
             if (checkInPiece.getId().equals(piece.getId()) && Boolean.TRUE.equals(checkInPiece.getCreateItem())) {
-              pieceFutures.add(purchaseOrderLineService.getOrderLineById(poLineId, requestContext)
-                .map(PoLineCommonUtil::convertToCompositePoLine)
-                .compose(compPol -> purchaseOrderStorageService.getPurchaseOrderByIdAsJson(compPol.getPurchaseOrderId(), requestContext)
-                  .map(HelperUtils::convertToCompositePurchaseOrder)
-                  .compose(purchaseOrder -> pieceCreateFlowInventoryManager.processInventory(purchaseOrder, compPol, piece, checkInPiece.getCreateItem(), requestContext))
-                .map(voidResult -> new PiecesHolder.PiecePoLineDto(poLineId, piece))));
+              pieceFutures.add(pieceCreateFlowInventoryManager.processInventory(holder.getPurchaseOrderPoLinePair().getKey(), holder.getPurchaseOrderPoLinePair().getValue(), piece, checkInPiece.getCreateItem(), requestContext)
```

Comment:
holder.getPurchaseOrderPoLinePair().getKey(),
holder.getPurchaseOrderPoLinePair().getValue()

can be moved to vaiables for easier reading